### DISCUSSION TITLE
Won't getting MeshAgent run on "our" Windows...

### DISCUSSION THREAD

**Discussion by simpsonetti** [Ylianst/MeshCentral#5615] [Q&A] [answered]
Hi, i'm testing MeshCentral. Therefor i have 2 VM with a regular Windows 10 LTSC installed, and then installed MEshAgent. So this is working as it should be. 

Then i have i testlaptop with "our" regular Windows10 with many regulations. So on this Laptopm i installed MeshAgent and it couldn't be connect to the Server. So I start manualy the MeshAgent.exe and it opens the MeshAgent. By clicking Connectiondetails there were shown the status, new version, "server-URL local" Server-ID, groupname and so on. 

If i click on "connect" it opens a command prompt with "unable to open database 1/10" but in the installfolder there is a meshagent.db file. Could this a permission problem??

So i installed wireshark to see what's going on there and by restarting the meshagent-service there will be no connection to my testserver. 
 Is there anything to figure out, whats the problem here with meshagent (or something else)?

Regards Sebastian

**Comment by si458** [comment]
1. the 'server-url' showing as 'local' means the laptop will only be able to connect to your meshcentral server IF its on the same network as the meshcentral server
2. the `unable to open database` means that the meshagent is running in the background already, so you cant open it again
how did you install the meshagent? did u download the exe, run it then click install? if you did then its running as a windows service in the background, so use the services tab in the 'Task manager' to stop/start the service etc

**Reply by simpsonetti** [reply]
Hi, number 2 first: i installed the exe downloaded from my server. The MeshAgentservice is running. Okay, that's the reason why. 

Number1: define Network: my meshcentral and the working agents are in 172.16.66.0/24 . The Laptop that isn't working is in 172.18.x.x and in an other vlan. BUT the network is working because it gets the proper ip, dns etc. 
I asked our networkguys and they told me, there is no firewall between the laptop and meshcentral, and the switches are routing all traffic ...

**Reply by si458** [reply, marked as answer]
the way `local` works is by sending our broadcast packets which get picked up by your meshcentral server,
so if you are in a completely different network range like you are, your laptop wont work, unless you relay the broadcast traffic from the laptop network to the meshcentral IP range,

the alternative you have is to change the server from LAN mode to WAN (OR LAN+WAN), and setting your cert name to a dns name that be resolved by BOTH your VMs and the LAPTOP (you can also try setting the certname to the meshcentral ip address but highly not recommended!) and then uninstalling the agent from the VMs+Laptop then reinstalling the new agent after you have restarted your meshcentral server

**Reply by simpsonetti** [reply]
okay, is there a way to get more information for that? i want to use DNS in a furthermore setup, but not now for my test. 
So how configure i the settings? Changing the values in meshcentral-data/config.json (wan true, lan true ) won't be used at a restart. So where yould i change this?

**Reply by simpsonetti** [reply]
Hi si458 , i've solved this. I think i have to watch all videos in order to get it working :)
Thats the video where it is described:
https://www.youtube.com/watch?v=gx5Fh3pQOns
